packages/InputGroup/InputGroup.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState, useEffect, useMemo, createRef } from 'react'
+import PropTypes from 'prop-types'
+import DecorativeIcon from '@tds/core-decorative-icon'
+import { Close } from '@tds/core-interactive-icon'
+import A11yContent from '@tds/core-a11y-content'
+import Box from '@tds/core-box'
+import Text from '@tds/core-text'
+import Strong from '@tds/core-strong'
+import { uniqueId } from '@tds/util-helpers'
+import { InputGroupStyle, LabelStyle } from './style'
+
+/**
+ * @version ./package.json
+ */
+const InputGroup = ({
+  id,
+  className,
+  placeholder,
+  defaultValue,
+  buttonText,
+  label,
+  hint,
+  onChange,
+  onKeyDown,
+  onButtonClick,
+  onClearButtonClick,
+}) => {
+  const inputRef = createRef()
+  const [inputValue, setInputValue] = useState('')
+  const inputId = useMemo(() => (label ? uniqueId(label) : id), [label, id])
+
+  useEffect(() => {
+    if (defaultValue) {
+      setInputValue(defaultValue)
+    }
+  }, [defaultValue])
+
+  return (
+    <Box className={className} between={1}>
+      {label && (
+        <LabelStyle htmlFor={inputId}>
+          <Text>
+            <Strong>{label}</Strong>
+          </Text>
+          <Text>{hint}</Text>
+        </LabelStyle>
+      )}
+      <InputGroupStyle hasValue={!!inputValue}>
+        <input
+          id={inputId}
+          type="text"
+          value={inputValue}
+          placeholder={placeholder}
+          onChange={e => {
+            setInputValue(e.target.value)
+            if (onChange) onChange(e)
+          }}
+          onKeyDown={onKeyDown}
+          ref={inputRef}
+        />
+        {!!inputValue && (
+          <Close
+            a11yText="Close"
```

Comment:
Please provide multi-lingual options for a11y text

Reply:
Changed